roles/v2/pool/src/lib/mining_pool/message_handler.rs

```diff
@@ -127,9 +134,55 @@ impl ParseDownstreamMiningMessages<(), NullDownstreamMiningSelector, NoRouting>
 
     fn handle_submit_shares_extended(
         &mut self,
-        _m: SubmitSharesExtended,
+        m: SubmitSharesExtended,
     ) -> Result<SendTo<()>, Error> {
-        todo!()
+        let res = self
+            .channel_factory
+            .safe_lock(|cf| cf.on_submit_shares_extended(m.clone()))
+            .unwrap();
+        match res {
+            Ok(res) => match res  {
+                roles_logic_sv2::channel_logic::channel_factory::OnNewShare::SendErrorDowsntream(m) => {
+                    Ok(SendTo::Respond(Mining::SubmitSharesError(m)))
+                }
+                roles_logic_sv2::channel_logic::channel_factory::OnNewShare::SendSubmitShareUpstream(_) => unreachable!(),
+                roles_logic_sv2::channel_logic::channel_factory::OnNewShare::RelaySubmitShareUpstream => unreachable!(),
+                roles_logic_sv2::channel_logic::channel_factory::OnNewShare::ShareMeetBitcoinTarget((share,t_id,coinbase)) => {
+                    info!("Found share that meet bitcoin target");
+                    let solution = SubmitSolution {
+                        template_id: t_id,
+                        version: share.get_version(),
+                        header_timestamp: share.get_n_time(),
+                        header_nonce: share.get_nonce(),
+                        coinbase_tx: coinbase.try_into().unwrap(),
```

Comment:
can this fail?

Reply:
do not remember max size of a bitcoin tx if is less than 64K bytes this can not fail.